Test/inline/testWhereExpr.bpl.expect

```diff
@@ -0,0 +1,59 @@
+
+procedure a();
+
+
+
+implementation a()
+{
+
+  anon0:
+    call b();
+    return;
+}
+
+
+
+procedure {:inline 1} b();
+
+
+
+implementation {:inline 1} b()
+{
+  var v: int where v > 3;
+
+  anon0:
+    havoc v;
+    return;
+}
+
+
+after inlining procedure calls
+procedure a();
+
+
+implementation a()
+{
+  var inline$b$0$v: int where inline$b$0$v > 3;
+
+  anon0:
+    goto inline$b$0$Entry;
+
+  inline$b$0$Entry:
+    havoc inline$b$0$v /* where inline$b$0$v > 3 */;
+    assume inline$b$0$v > 3;
```

Comment:
It seems that this assume is injected automatically by the inlining implementation. Is that correct?

Reply:
Yes, I think [these lines](https://github.com/boogie-org/boogie/blob/1969ad0f9b8c8d5de4aa5f8021d2364e4fcb366a/Source/Core/Inline.cs#L657-L688) are responsible for doing this. The justification for adding `havoc` is that the procedure might be invoked in a loop and so the values of its local variables would have to be reset on each iteration. There are `FIXME` comments surrounding the  added `assume` and I think they might be technically redundant but I am not confident enough in the semantics.